### PR TITLE
chore: split ARRAYCONTAINS into JSON_ARRAY_CONTAINS and ARRAY_CONTAINS

### LINKED ISSUE #3729: Breaking change: Rename `ARRAYCONTAINS`
The `ARRAYCONTAINS` method has the description: 

> Given JSON or AVRO array checks if a search  value contains in it 

Looking at the code it seems to have two different code paths:

1. `ARRAYCONTAINS(STRING, STRING)`: If the first parameter passed is a `STRING` it expects it to be a JSON array, e.g. a string such as `"[1,2,3]"`. It will return true if that array contains an element that matches the second parameter, which must also be a string.
1. `ARRAYCONTAINS(ARRAY<T>, T)`, where `T` can be `STRING`, `INT`, `BIGINT` or `DOUBLE` only. If the first parameter passed is an `ARRAY` it looks for the second parameter in the supplied array.

Note: the second code path was added later. The initial version only handled JSON arrays.

Note: there is no support for any 'Avro Array' as the description states and the description does not cover the second code path.

In short, this is two methods rolled into one: a method for checking if an `ARRAY` contains a certain element, and an a method for checking if a Json array contains a certain element.  

We should split this into two methods:

1. `ARRAYCONTAINS` should check for the presence of an element in an `ARRAY` instance. It should be extended to work with _any_ element type, i.e. extended to support `BOOLEAN`, `DECIMAL`, `MAP`, `ARRAY` and `STRUCT` element types.
1. `JSONARRAYCONTAINS` which should accept a `STRING` containing a JSON array. It should be extended to work with element types other than `STRING`. (It looks like the code in `ArrayContainsKudf` can already handle other types, but the only combination registered in `InternalFunctionRegistry` is `ARRAYCONTAINS(STRING, STRING)`.  e.g. we should support `JSONARRAYCONTAINS(STRING, INT)`, and variants with second parameter of type `BIGINT`, `BOOLEAN`, `DOUBLE`, `MAP`, `ARRAY` `DECIMAL` and `STRUCT`.

### PR DESCRIPTION
**Pull request by agavra**
BREAKING CHANGE: the ARRAYCONTAINS function now needs to be referenced
as either JSON_ARRAY_CONTAINS or ARRAY_CONTAINS depending on the
intended param types

### Description 
fixes #3729 - renames and does some cleanup around the ARRAYCONTAINS functionality in KSQL

### Testing done 

- extra unit and qtt tests

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        assertEquals(true, jsonUdf.contains("[[222222222222222], 33333]", 33333L));
+        assertEquals(true, jsonUdf.contains("[{}, \"abc\", null, 1]", 1L));
+        assertEquals(false, jsonUdf.contains("[[222222222222222], 33333]", 222222222222222L));
+        assertEquals(false, jsonUdf.contains("[{}, \"abc\", null, [1]]", 1L));
+        assertEquals(false, jsonUdf.contains("[{}, \"abc\", null, {\"1\":1}]", 1L));
+    }
+
+    @Test
+    public void shouldFindDoublesInJsonArray() {
+        assertEquals(true, jsonUdf.contains("[-1.0, 2.0, 3.0]", 2.0));
```
Comment (vcrfxia) [inline ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java]: What about
```
jsonUdf.contains("[-1.0, 2, 3.0]", 2.0)
```
and
```
jsonUdf.contains("[-1.0, 2.0, 3.0]", 2)
```
? 

I assume both return false?

File: ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2018 Confluent Inc.
```
Comment (vcrfxia) [inline ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java]: nit: year

Comment (agavra) [reply ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java]: Almost 2020! Gotta change my scripts :D

File: ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java (hunk 1 above)
Comment (agavra) [reply ksql-engine/src/test/java/io/confluent/ksql/function/udf/json/JsonArrayContainsTest.java]: There's a test for this at the end of `shouldFindDoublesInJsonArray` but I added an extra one for longs.

**Pull request merged**